src/v1/components/data-display/UserProfile/UserProfile.tsx

```diff
@@ -25,11 +25,22 @@ const UserProfile: React.FC<UserProfileProps> = ({ fullName, children }) => {
     setAnchorEl(null);
   };
 
+  const usernameStyle = css`
+    display: none;
+    @media (min-width: 768px) {
+      display: block;
+      max-width: 11rem;
+    }
+    @media (min-width: 1024px) {
+      max-width: 18rem;
+    }
+  `;
+
   return (
     <MUIStack direction="row" spacing={1} alignItems="center">
       <H2
         variant="h6"
-        className="username"
+        css={usernameStyle}
```

Comment:
Hm. Not sure about css-to-js. Its perhaps ok - but I'm unsure.

For now lets stick to these "MUI blessed" methods of styling (in order of consideration):

1. [One-off customization](https://mui.com/material-ui/customization/how-to-customize/#1-one-off-customization)
1. [Reusable component](https://mui.com/material-ui/customization/how-to-customize/#2-reusable-component)
1. [Global theme overrides](https://mui.com/material-ui/customization/how-to-customize/#3-global-theme-overrides)
1. [Global CSS override](https://mui.com/material-ui/customization/how-to-customize/#4-global-css-override)

So perhaps based on [response values](https://mui.com/system/getting-started/usage/#responsive-values)

```txx
<H2
sx={{
  display: { xs: 'none', sm: 'block' },
  maxWidth: {
    sm: '11rem',  // Applies when the screen width is 600px or more
    md: '18rem'   // Applies when the screen width is 960px or more
  }
}}
```

Reply:
Done.